Re-wrote branding overview, adding more info around exection context (#829)
gravitee-io/gravitee-docs: pages/am/3.x/user-guide/branding/user-guide-branding.adoc

:page-folder: am/user-guide
:page-layout: am

== Overview

> Since the dawn of centralized IAM, moderns apps have moved away from login directly in the app. This means however that it is critical that you offer a consistent user experience when redirecting users to the login scenarios in AM. The user should feel confindent and secure. An inconsitent or unclear user experience might affect customer retention and perception of your company.

Reviewer: ```suggestion
Since the dawn of centralized IAM, modern apps have moved away from logins directly in the app, so it is critically important to offer a consistent user experience when redirecting users to the login scenarios in AM. Users should feel confindent and secure and an inconsistent or unclear user experience might affect your organization's customer retention and reputation
```

> AM allows you to customize the look and feel of end user forms displayed to the user in the various flows such as login, reset password and registering user. AM also allows you to customize the look adn feel of emails being sent out to end user. This gives you full flexibility to apply the relevant graphical user experience.

Reviewer 2: ```suggestion
AM allows you to customize the look and feel of end user forms displayed to the user in the various flows such as login, reset password and registering user. AM also allows you to customize the look and feel of emails being sent out to end user. This gives you full flexibility to apply the relevant graphical user experience.
```

> AM allows you to customize the look and feel of end user forms displayed to the user in the various flows such as login, reset password and registering user. AM also allows you to customize the look and feel of emails being sent out to end user. This gives you full flexibility to apply the relevant graphical user experience.

Reviewer: ```suggestion
AM enables you to customize the look and feel of the end-user forms displayed in the various flows - such as login, password reset, and user registration. AM also allows you to customize the look and feel of emails that are sent out to end users, giving you full flexibility to apply the relevant graphical user experience.
```

> Extending the branding AM also allows you to leverage context variables and use these in your forms. You'll find more on how to unleash this power under Execution context sections linked below.

Reviewer: ```suggestion
Extending the branding AM also enables you to leverage context variables and use these in your forms. You can find more information on how to unleash this power in the 'Execution context' sections listed below.
```

* link:{{ '/am/current/am_userguide_user_management_forms.html#execution_context' | relative_url }}[Execution context for Custom pages]
* link:{{ '/am/current/am_userguide_branding_email_templates.html#execution_context' | relative_url }}[Execution context for Custom email templates]

== Branding on different levels

> To fully tend to your applications branding and user experience AM allows you to either customize forms on link:{{ '/am/current/am_userguide_security_domains.html' | relative_url }}[Security domain level] or on link:{{ '/am/current/am_userguide_clients.html' | relative_url }}[Application level]. Some application can rely on your overall company branding while other may be in need of custom branding. All to make sure that you can create the best experience to your users regardless of application.

Reviewer: ```suggestion
To fully tend to your application branding and user experience, AM enables you to customize forms either on link:{{ '/am/current/am_userguide_security_domains.html' | relative_url }}[Security domain level] or on link:{{ '/am/current/am_userguide_clients.html' | relative_url }}[Application level]. This addresses cases where some applications can be fine with relying on your overall company branding but others may be in need of a more custom branding touch.
```
